File: Java/src/PageFactory/Residence.java
package PageFactory;

import org.openqa.selenium.By;
import org.openqa.selenium.JavascriptExecutor;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.PageFactory;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.util.concurrent.TimeUnit;

public class Residence
{
    WebDriver driver;
    WebDriverWait wait;

    public Residence(WebDriver driver, WebDriverWait wait)
    {
        this.driver = driver;
        this.wait = wait;
        PageFactory.initElements(driver, this);
    }

    public void ResidenceStatus(String status) throws InterruptedException
    {
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//button[@type=\"button\"]"))).click();
        TimeUnit.SECONDS.sleep(2);
        driver.findElement(By.xpath("//div[text()=\"" + status + "\"]")).click();
        TimeUnit.SECONDS.sleep(2);
    }

    public void Address(String address) throws InterruptedException
    {
        TimeUnit.SECONDS.sleep(2);
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("aginputaddress")));
        driver.findElement(By.id("aginputaddress")).sendKeys(address);
    }

    public void City(String city) throws InterruptedException {
        TimeUnit.SECONDS.sleep(2);
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("aginputcity")));
        Actions actions = new Actions(driver);
        WebElement element = driver.findElement(By.id("aginputcity"));
        actions.moveToElement(element);
        actions.click();
        actions.sendKeys(city);
        actions.build().perform();
    }

    public void State(String state) throws InterruptedException
    {
        TimeUnit.SECONDS.sleep(2);
        driver.findElements(By.xpath("//button[@type=\"button\"]")).get(1).click();
        TimeUnit.SECONDS.sleep(2);
        driver.findElement(By.xpath("//div[text()=\"" + state + "\"]")).click();
    }

    public void Zipcode(String zipcode) throws InterruptedException
    {
        TimeUnit.SECONDS.sleep(2);
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("aginputzip")));
        driver.findElement(By.id("aginputzip")).sendKeys(zipcode);
    }

    public void MoveInDate(String move_in_date) throws InterruptedException {
        TimeUnit.SECONDS.sleep(2);
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("aginputmoveindate")));
        driver.findElement(By.id("aginputmoveindate")).sendKeys(move_in_date);
    }

    public void MortgagePayment(String mortgage_payment) throws InterruptedException {
        TimeUnit.SECONDS.sleep(2);
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("aginputpayment")));
        driver.findElement(By.id("aginputpayment")).sendKeys(mortgage_payment);
    }

    public void Navigate() throws InterruptedException
    {
        ((JavascriptExecutor) driver).executeScript("window.scrollTo(0, document.body.scrollHeight)");
        TimeUnit.SECONDS.sleep(8);
        //wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//button[text()=\"NEXT\"]")));
        Actions actions = new Actions(driver);
        WebElement element = driver.findElement(By.xpath("//button[@type=\"submit\"]"));
        actions.moveToElement(element);
        actions.click();
        actions.build().perform();

    }
}

File: Python/residence.py
import time
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

class Residence(object):
    def __init__(self, driver):
        self.driver = driver
        self.wait = WebDriverWait(driver, 30)

    # Selects recidence status from dropdown
    def ResidenceStatus(self, status):
        self.driver.find_elements_by_xpath('//button[@type="button"]')[0].click()
        #self.driver.find_element_by_css_selector('#agselecthouseinfo > div:nth-child(1) > button').click()
        time.sleep(2)
        self.driver.find_element_by_xpath('//div[text()="' + status + '"]').click()
        time.sleep(2)

    # Inputs Address
    def Address(self, address):
        self.wait.until(EC.visibility_of_element_located((By.ID, 'aginputaddress')))
        self.driver.find_element_by_id('aginputaddress').send_keys(address)

    # Inputs City
    def City(self, city):
        self.driver.find_element_by_id('aginputcity').send_keys(city)

    # Inputs State
    def State(self, state):
        time.sleep(2)
        self.driver.find_elements_by_xpath('//button[@type="button"]')[1].click()
        #self.driver.find_element_by_css_selector('#agselectstate > div:nth-child(1) > button').click()
        time.sleep(2)
        self.driver.find_element_by_xpath('//div[text()="' + state + '"]').click()

    # Inputs Zipcode
    def Zipcode(self, zipcode):
        time.sleep(2)
        self.wait.until(EC.visibility_of_element_located((By.ID, 'aginputzip')))
        self.driver.find_element_by_id('aginputzip').send_keys(zipcode)

    # Inputs Move in date
    def MoveInDate(self, move_in_date):
        self.driver.find_element_by_id('aginputmoveindate').send_keys(move_in_date)

    # Inputs Mortgage Payment
    def MortgagePayment(self, mortgage_payment):
        self.driver.find_element_by_id('aginputpayment').send_keys(mortgage_payment)

    # Clicks on button "Next"
    def Navigate(self):
        self.driver.save_screenshot('Residence_Information.png')
        time.sleep(2)
        self.driver.execute_script("window.scrollTo(0, document.body.scrollHeight);")
        self.wait.until(EC.element_to_be_clickable((By.XPATH, '//button[text()="Next"]')))
        self.driver.find_element_by_xpath('//button[text()="Next"]').click()

